kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -194,11 +186,26 @@ def __init__(self, name, init_config, instances):
         pod_level_metrics = inst.get('pod_level_metrics', DEFAULT_POD_LEVEL_METRICS)
         self.pod_level_metrics = ["{0}.{1}".format(self.NAMESPACE, x) for x in pod_level_metrics]
 
-        kubelet_instance = self._create_kubelet_prometheus_instance(inst)
-        probes_instance = self._create_probes_prometheus_instance(inst)
+        # configuring the different instances use to scrape the 3 kubelet endpoints
+        prom_url, get_prom_url_err = get_prometheus_url("dummy_url/cadvisor")
+
+        cadvisor_instance = self._create_cadvisor_prometheus_instance(inst, prom_url)
+        if len(inst.get('ignore_metrics', {})) > 0:
+            # Add entries from configuration to ignore_metrics in the cadvisor collector.
+            cadvisor_instance['ignore_metrics'].extend(
+                m for m in inst.get('ignore_metrics', {}) if m not in cadvisor_instance['ignore_metrics']
+            )
+
+        kubelet_instance = self._create_kubelet_prometheus_instance(inst, prom_url)
+        probes_instance = self._create_probes_prometheus_instance(inst, prom_url)
         generic_instances = [cadvisor_instance, kubelet_instance, probes_instance]
+
         super(KubeletCheck, self).__init__(name, init_config, generic_instances)
 
+        # we need to wait that `super()` was exectued to have the self.log instance created
+        if get_prom_url_err:
+            self.log.Warning('get_prometheus_url() failed: %s', get_prom_url_err)
```

Comment:
```suggestion
            self.log.warning('failed to query kubelet: %s', get_prom_url_err)
```
IIRC `warning` should not be capitalized. Also I think the error message could be more indicative